src/jdk.compiler/share/classes/jdk/internal/shellsupport/doc/JavadocHelper.java

```diff
@@ -495,7 +495,7 @@ public Void scan(DocTree tree, Void p) {
                             //if there is a newline immediately behind this tree, insert behind
                             //the newline:
                             long endPos = sp.getEndPosition(null, dcTree, tree);
-                            if (endPos >= 0) {
+                            if (endPos >= offset) {
```

Comment:
What's that about?

Reply:
Without the change, you can get an exception like `StringIndexOutOfBoundsException` for a bad `endPos`.

I believe there's some underlying weirdness for synthesized HTML comments (i.e. full `<html>...</html>` comments e.g. for `package.html`) that needs to be investigated, but this is in `jshell` code and getting out of scope for this round of work.